great_expectations/data_context/data_context.py

```diff
@@ -457,7 +457,12 @@ def _init_datasources(self, config):
                 # this error will happen if our configuration contains datasources that GE can no longer connect to.
                 # this is ok, as long as we don't use it to retrieve a batch. If we try to do that, the error will be
                 # caught at the context.get_batch() step. So we just pass here.
-                pass
+                if self._ge_cloud_mode:
+                    # when running in cloud mode, we want to know if a datasource has been improperly configured at
+                    # init time.
+                    raise
+                else:
+                    pass
```

Comment:
I'm adding some logging around the non-ge_cloud_mode path in another PR so there may be some overlap